neureka/hal/neureka_task.c

```diff
@@ -133,6 +133,10 @@ void neureka_task_set_ptrs_conv(neureka_task_t *task, uint32_t input_ptr,
   task->data.infeat_ptr =
       neureka_pad_ptr(input_ptr, w_in, w_in_stride, padding_top, padding_left);
   task->data.outfeat_ptr = output_ptr;
+  if ((task->data.cfg.conf0 & NEUREKA_MASK_FLAG_WEIGHT_SOURCE) ==
+      NEUREKA_FLAG_WEIGHT_SOURCE_WMEM) {
+    weights_ptr -= 0x10400000;
```

Comment:
I would not modify the pointer in place; Storing a dedicated offset variable would be nicer, or, alternatively, ensure that the `weights_ptr` is only decremented if it's larger than `0x10400000`

Reply:
Something along the lines of:
`task->data.weights_ptr = weights_ptr - 0x10400000;`
Does the same thing, but if you call the function twice you'll get the same result since the `weights_ptr` is not modified in place.

Reply:
Ah, I see now why you are confused. `weights_ptr` is not actually a pointer but an immediate. Maybe rename to weights_ptr_addr to make this clearer.

Reply:
Changed here d3b18d6769c8527c8831c9e987e5c439292c7372